src/gc/pooltable.d

```diff
@@ -0,0 +1,287 @@
+/**
+ * A sorted array to quickly lookup pools.
+ *
+ * Copyright: Copyright Digital Mars 2001 -.
+ * License:   $(WEB www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+ * Authors:   Walter Bright, David Friedman, Sean Kelly, Martin Nowak
+ */
+module gc.pooltable;
+
+static import cstdlib=core.stdc.stdlib;
+
+struct PoolTable(Pool)
+{
+    import core.stdc.string : memmove;
+
+nothrow:
+    void reset()
+    {
+        cstdlib.free(pools);
+        pools = null;
+        npools = 0;
+    }
+
+    bool insert(Pool* pool)
+    {
+        auto newpools = cast(Pool **)cstdlib.realloc(pools, (npools + 1) * pools[0].sizeof);
+        if (!newpools)
+            return false;
+
+        pools = newpools;
+
+        // Sort pool into newpooltable[]
+        size_t i;
```

Comment:
IMO assuming a default value is intended to get consistent results even if uninitialized, not to avoid writing an initializer. float and char would have more "useful" defaults otherwise.
Not a blocker, though.